pkg/cmd/roachtest/roachtestutil/mixedversion/planner.go

```diff
@@ -235,13 +281,31 @@ func (p *testPlanner) changeVersionSteps(
 		previousVersionNodes[i], previousVersionNodes[j] = previousVersionNodes[j], previousVersionNodes[i]
 	})
 
+	// waitProbability is the probability that we will wait and allow
+	// the cluster to stay in the current state for a while, in case we
+	// are not scheduling user hooks.
+	const waitProbability = 0.5
+	waitIndex := -1
+	if !scheduleHooks && p.prng.Float64() < waitProbability {
+		waitIndex = p.prng.Intn(len(previousVersionNodes))
+	}
+
 	var steps []testStep
-	for _, node := range previousVersionNodes {
+	for j, node := range previousVersionNodes {
 		steps = append(steps, p.newSingleStep(
 			restartWithNewBinaryStep{version: to, node: node, rt: p.rt, settings: p.clusterSettings()},
 		))
 		p.currentContext.changeVersion(node, to)
-		steps = append(steps, p.hooks.MixedVersionSteps(p.currentContext)...)
+		if scheduleHooks {
+			steps = append(steps, p.hooks.MixedVersionSteps(p.currentContext)...)
+		} else if j == waitIndex {
+			// If we are not scheduling user-provided hooks, we wait a short
+			// while in this state to allow some time for background
+			// operations to run.
+			possibleWaitMinutes := []int{5, 10, 15}
```

Comment:
15 mins may seem on a higher end for all upgrades, no?

Reply:
Yeah, I think it's probably fine on master, but it could add up to quite a bit in the worst case scenario in release branches (especially 23.1). I changed this to `{1,5,10}`; we can reduce it further if necessary in the future.